dspace-server-webapp/src/test/java/org/dspace/app/rest/ResourcePolicyRestRepositoryIT.java

```diff
@@ -0,0 +1,577 @@
+                .withUser(admin).build();
+        context.restoreAuthSystemState();
+
+        String authToken = getAuthToken(admin.getEmail(), password);
+        getClient(authToken).perform(get("/api/authz/resourcepolicies")).andExpect(status().isMethodNotAllowed());
+    }
+
+    @Test
+    public void findAllUnAuthenticatedTest() throws Exception {
+
+        context.turnOffAuthorisationSystem();
+        Community community = CommunityBuilder.createCommunity(context).withName("My community").build();
+
+        ResourcePolicyBuilder.createResourcePolicy(context).withDspaceObject(community).withAction(Constants.READ)
+                .withUser(admin).build();
+        context.restoreAuthSystemState();
+
+        getClient().perform(get("/api/authz/resourcepolicies")).andExpect(status().isUnauthorized());
+    }
+
+    @Test
+    public void findOneTest() throws Exception {
+        context.turnOffAuthorisationSystem();
+
+        Community community = CommunityBuilder.createCommunity(context).withName("My community").build();
+
+        ResourcePolicy resourcePolicy = ResourcePolicyBuilder.createResourcePolicy(context).withDspaceObject(community)
+                .withAction(Constants.READ).withPolicyType(ResourcePolicy.TYPE_CUSTOM).withUser(admin).build();
+
+        context.restoreAuthSystemState();
+
+        String authToken = getAuthToken(admin.getEmail(), password);
+        getClient(authToken).perform(get("/api/authz/resourcepolicies/" + resourcePolicy.getID()))
+                .andExpect(status().isOk()).andExpect(content().contentType(contentType))
+                .andExpect(jsonPath("$", is(
+                        ResoucePolicyMatcher.matchResourcePolicy(resourcePolicy)
+                        )))
+                .andExpect(jsonPath("$._links.self.href", Matchers
+                        .containsString("/api/authz/resourcepolicies/" + resourcePolicy.getID())));
+    }
+
+    @Test
+    public void findOneUnAuthenticatedTest() throws Exception {
+        context.turnOffAuthorisationSystem();
+        Community community = CommunityBuilder.createCommunity(context).withName("My community").build();
+
+        ResourcePolicy resourcePolicy = ResourcePolicyBuilder.createResourcePolicy(context).withDspaceObject(community)
+                .withAction(Constants.READ).withPolicyType(ResourcePolicy.TYPE_CUSTOM).withUser(eperson).build();
+
+        context.restoreAuthSystemState();
+
+        getClient().perform(get("/api/authz/resourcepolicies/" + resourcePolicy.getID()))
+                .andExpect(status().isUnauthorized());
+
+    }
+
+    @Test
+    public void findOneNotFoundTest() throws Exception {
+
+        String authToken = getAuthToken(admin.getEmail(), password);
+        getClient(authToken).perform(get("/api/authz/resourcepolicies/" + UUID.randomUUID().toString()))
+        .andExpect(status().isNotFound());
+    }
+
+
+    @Test
+    public void searchOneResoucesPolicyByEpersonUuidTest() throws Exception {
```

Comment:
better to keep the test names aligned with the repository names. Replace search with find